lib/metadata.js

```diff
@@ -70,8 +70,8 @@ class Metadata {
     constructor(source) {
         if (!source) {
             throw new GenericError(`No source provided`);
-        } else if (!source.name && !source.path && !source.mimetype) {
-            throw new GenericError(`Provided source must contain one of 'name', 'path', or 'mimetype' fields`);
+        } else if (!source.name && !source.path && !source.type) {
+            throw new GenericError(`Provided source must contain one of 'name', 'path', or 'type' fields`);
```

Comment:
I assume that this lib may be used in other places, so the requirement itself is whenever it contains at least one of those. type happens to be here since it is a requirement of refinePlan.